app/collections/components/AdminEditors.tsx

```diff
@@ -0,0 +1,150 @@
+import { getAlgoliaResults } from "@algolia/autocomplete-js"
+import { MembershipRole } from "@prisma/client"
+import algoliasearch from "algoliasearch"
+import Autocomplete from "app/core/components/Autocomplete"
+import SearchResultWorkspace from "app/core/components/SearchResultWorkspace"
+import DeleteEditorModal from "app/core/modals/DeleteEditorModal"
+import SetEditorToInactiveModal from "app/core/modals/SetEditorToInactiveModal"
+import UpgradeCollectionModal from "app/core/modals/UpgradeCollectionModal"
+import { Link, Routes, useMutation } from "blitz"
+import toast from "react-hot-toast"
+import addEditor from "../mutations/addEditor"
+import changeEditorRole from "../mutations/changeEditorRole"
+
+const searchClient = algoliasearch(process.env.ALGOLIA_APP_ID!, process.env.ALGOLIA_API_SEARCH_KEY!)
+
+const EditorCard = ({ editor, isAdmin, isSelf, refetchFn }) => {
+  const [changeEditorRoleMutation] = useMutation(changeEditorRole)
+
+  return (
+    <>
+      <div className={`flex ${editor.isActive ? "" : "opacity-50"} my-2`}>
+        <img src={editor.workspace.avatar} className="mx-2 h-12 w-12 rounded-full" />
+        <div className="inline-block flex-grow">
+          <Link href={Routes.HandlePage({ handle: editor.workspace.handle })}>
+            <a target="_blank">
+              <p className="line-clamp-1">
+                {editor.workspace.firstName} {editor.workspace.lastName}
+              </p>
+              <p className="text-sm">@{editor.workspace.handle}</p>
+            </a>
+          </Link>
+        </div>
+        {isAdmin && (
+          <>
+            <select
+              onChange={(info) => {
+                toast.promise(
+                  changeEditorRoleMutation({ editorId: editor.id, role: info.target.value }),
+                  {
+                    loading: `Changing role to ${info.target.value.toLowerCase()}...`,
+                    success: () => {
+                      refetchFn()
+                      return `Changed role to ${info.target.value.toLowerCase()}!`
+                    },
+                    error: (err) => {
+                      return `${err}`
+                    },
+                  }
+                )
+              }}
+              defaultValue={editor.role}
+              className="placeholder-font-normal block appearance-none rounded-md border border-gray-400 bg-white px-4 py-2 pr-6 text-sm font-normal placeholder-gray-400 shadow-sm focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-0 dark:border-gray-600 dark:bg-transparent dark:text-gray-200 "
+            >
+              {Object.values(MembershipRole).map((role) => {
+                return <option key={role}>{role}</option>
+              })}
+            </select>
```

Comment:
I noticed that I can still select a role that's not allowed (for example, being a user for my own collection). 

![image](https://user-images.githubusercontent.com/17035406/191788330-70ed9b58-cc90-4eef-919a-e52ba35d4a19.png)

We don't have to fix this for this PR, but something we want to address later? I think we can make a stateful value and make this dropdown controlled to manage the value (so that the selection goes back to "OWNER" even the user selects "USER")

Reply:
Yes! I was doing some hacking of this but there's 100% improvements - could you open that issue to track it? 👍

Reply:
Sounds good—here's the opened issue! https://github.com/libscie/ResearchEquals.com/issues/742